spec.html

```diff
@@ -36046,8 +36102,11 @@ <h1>
             1. Assert: _parseResult_ is a |Pattern| Parse Node.
             1. Set _obj_.[[OriginalSource]] to _P_.
             1. Set _obj_.[[OriginalFlags]] to _F_.
-            1. NOTE: The definitions of _DotAll_, _IgnoreCase_, _Multiline_, and _Unicode_ in <emu-xref href="#sec-notation"></emu-xref> refer to this value of _obj_.[[OriginalFlags]].
-            1. Set _obj_.[[RegExpMatcher]] to CompilePattern of _parseResult_.
+            1. Let _rer_ be the RegExp Record { [[IgnoreCase]]: _i_, [[Multiline]]: _m_, [[DotAll]]: _s_, [[Unicode]]: _u_ }.
```

Comment:
I am not a fan of constructing a record with missing fields, especially since we're in fact _using_ the record before it is fully initialized (on the next line).

I would much prefer to fully initialize `_rer_` in a single step. This will require refactoring `WordCharacters` and `Canonicalize` to take `unicode` and `ignoreCase` booleans explicitly, which I don't see a problem with.

Reply:
I tried your suggestion (refactoring `WordCharacters` and `Canonicalize`), but I wasn't happy with the results. Instead, I eliminated the `[[WordCharacters]]` field, which also eliminates the call to `WordCharacters` with an unfinished `_rer_`. This makes it easy to fully initialize `_rer_` in a single step.

(Would you prefer if we called `CountLeftCapturingParensWithin(_parseResult_)` in a preliminary step?)

Reply:
> I eliminated the `[[WordCharacters]]` field

Works for me. [This proposal](https://github.com/tc39/proposal-regexp-modifiers) (which allows changing some flags in the middle of a regex) needs to do the same thing anyway.

> (Would you prefer if we called `CountLeftCapturingParensWithin(_parseResult_)` in a preliminary step?)

That does seem like it would be a (marginal) improvement, yeah.